src/lib/api/entity-api.ts

```diff
@@ -362,15 +362,16 @@ export namespace EntityApi {
    * and handles cache invalidation.
    *
    * @template T The entity type
+   * @template U The entity type returned by the API
    * @param baseUrl The base URL for the entity API endpoint
    * @param api Object containing CRUD operations for the entity
    * @returns Object with CRUD methods, loading state, and error state
    */
-  export const useEntityMutation = <T>(
+  export const useEntityMutation = <T, U>(
     baseUrl: string,
     api: {
-      create: (entity: T) => Promise<T>;
-      createNested: (id: Id, entity: T) => Promise<T>;
+      create: (entity: T) => Promise<U>;
+      createNested: (id: Id, entity: T) => Promise<U>;
```

Comment:
This is where I changed this to allow us to provide differing types for what is sent vs what is received

Reply:
I like this change very much, great addition!

Reply:
Really like this  change - you could even have U default to T:

```
export const useEntityMutation = <T, U = T>(
```

because I see several places where they are still the same, but since we require two generics you have to assign two of the same types. Unless we envision that being more uncommon than it is now.